### PR TITLE
TST: raise InvalidIndexError with IntervalIndex.get_value and get_loc

### LINKED ISSUE #31117: REF: IntervalIndex.get_value partial slicing
In trying to refactor IntervalIndex.get_loc/get_value to more closely follow the other implementations, I stumbled on a behavior in a test case that surprised me:

```
idx = IntervalIndex.from_tuples([(1, 5), (3, 7)])
s = Series(range(len(idx)), index=idx)
key = [4]

>>> idx.get_loc(key)  # <-- behaves like all the other get_locs in requiring scalar
KeyError: [4]

>>> idx.get_value(s, key)   # <-- surprising
(1, 5]    0
(3, 7]    1
dtype: int64
```

All of the other FooIndex.get_value methods raise InvalidIndexError on non-scalar.

AFAICT this is behaving sort of like a slicing operation, since 4 is in the interior of both of these intervals.  @jschendel is that correct?  If so, we should try to refactor IntervalIndex.get_value to explicitly dispatch for this case, kind of like DatetimeIndex._get_string_slice

### PR DESCRIPTION
**Pull request by primaprashant**
ensure InvalidIndexError is raised when non-scalar is passed as key to
get_loc and get_value methods for IntervalIndex

- [X] closes #31117 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/indexes/interval/test_interval.py [hunk 1]
```diff
@@ -857,6 +858,37 @@ def test_is_all_dates(self):
         year_2017_index = pd.IntervalIndex([year_2017])
         assert not year_2017_index.is_all_dates
 
+    def test_get_loc_errors(self):
```
Comment (jbrockmendel) [inline pandas/tests/indexes/interval/test_interval.py:861]: this should go in tests.indexes.interval.test_indexing.TestGetLoc

Comment (prashant616) [reply pandas/tests/indexes/interval/test_interval.py]: @jbrockmendel thanks for the review, moved the tests.

File: pandas/tests/indexes/interval/test_indexing.py [hunk 2]
```diff
@@ -158,6 +158,20 @@ def test_get_loc_decreasing(self, values):
         expected = 0
         assert result == expected
 
+    def test_get_loc_non_scalar_errors(self):
```
Comment (jschendel) [inline pandas/tests/indexes/interval/test_indexing.py:161]: Can you parametrize this test over `key` so we only need the `pytest.raises` logic once?  Also, can you move the comment with the github issue number to be the first thing in the test, e.g.
```python
@pytest.mark.parametrize("key", [[5], (2, 3)])
def test_get_value_errors(self, key):
    # GH 31117
    idx = ...
```

File: pandas/tests/indexes/interval/test_interval.py [hunk 3]
```diff
@@ -857,6 +858,22 @@ def test_is_all_dates(self):
         year_2017_index = pd.IntervalIndex([year_2017])
         assert not year_2017_index.is_all_dates
 
+    def test_get_value_errors(self):
```
Comment (jschendel) [inline pandas/tests/indexes/interval/test_interval.py:861]: Same thing regarding parametrize and the github issue number.

File: pandas/tests/indexes/interval/test_interval.py [hunk 4]
```diff
@@ -857,6 +858,22 @@ def test_is_all_dates(self):
...
+        idx = IntervalIndex.from_tuples([(1, 3), (2, 4), (3, 5), (7, 10), (3, 10)])
+        s = pd.Series(range(len(idx)), index=idx)
+
+        # key is not scalar
+        # GH 31117
+        key = [5]
+        msg = str(key)
+        with pytest.raises(InvalidIndexError, match=msg):
+            idx.get_value(s, key)
```
Comment (jschendel) [inline pandas/tests/indexes/interval/test_interval.py:870]: `get_value` was recently deprecated in #33907, so you'll need to merge master (`git merge upstream/master`) and then ensure that we catch the `FutureWarning` associated with the deprecation:
```python
with pytest.raises(InvalidIndexError, match=msg):
    with tm.assert_produces_warning(FutureWarning):
        idx.get_value(s, key)
```

File: pandas/tests/indexes/interval/test_indexing.py (hunk 2 above)
Comment (prashant616) [reply pandas/tests/indexes/interval/test_indexing.py]: changed

File: pandas/tests/indexes/interval/test_interval.py (hunk 3 above)
Comment (prashant616) [reply pandas/tests/indexes/interval/test_interval.py]: changed

File: pandas/tests/indexes/interval/test_interval.py (hunk 4 above)
Comment (prashant616) [reply pandas/tests/indexes/interval/test_interval.py]: changed

**Comment by jreback** [comment]
thanks @prashant616

**Pull request merged**